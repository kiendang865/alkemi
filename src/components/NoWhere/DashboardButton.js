import React from "react";
import PropTypes from "prop-types";

import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

const styles = (theme) => ({
    button: {
        margin: theme.spacing(3),
    },
});

const NoWhereDashboardButton = ({ classes, onClick }) => (
    <Button
        variant="raised"
        color="default"
        className={classes.button}
        onClick={onClick}
    >
        Bring me back home!
    </Button>
);

NoWhereDashboardButton.propTypes = {
    classes: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default withStyles(styles)(NoWhereDashboardButton);
