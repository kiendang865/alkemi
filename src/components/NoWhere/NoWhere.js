import React from "react";
import PropTypes from "prop-types";

import ReportIcon from "@material-ui/icons/Report";
import { withStyles } from "@material-ui/core/styles";
import { grey } from "@material-ui/core/colors";

import NoWhereDashboardButton from "../../containers/NoWhere/DashboardButton";

const styles = (theme) => ({
    nowhere: {
        textAlign: "center",
    },
    nowhereHeader: {
        padding: theme.spacing(3),
    },
    nowhereTitle: {
        fontSize: "1.5em",
        color: grey[500],
    },
    nowhereIcon: {
        height: 80,
        width: 80,
        color: grey[500],
    },
});

const NoWhere = ({ classes }) => (
    <div className={classes.nowhere}>
        <div className={classes.nowhereHeader}>
            <ReportIcon className={classes.nowhereIcon} />
            <h1 className={classes.nowhereTitle}>
                Oops... It seems that you are lost!
            </h1>
        </div>
        <NoWhereDashboardButton />
    </div>
);

NoWhere.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(NoWhere);
