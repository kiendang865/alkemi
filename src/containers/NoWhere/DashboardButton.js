import { connect } from "react-redux";
import { push } from "connected-react-router";

import NoWhereDashboardButton from "components/NoWhere/DashboardButton";
import { DASHBOARD } from "constants/routes";

const mapDispatchToProps = (dispatch) => ({
    onClick: () => dispatch(push(DASHBOARD)),
});

export default connect(undefined, mapDispatchToProps)(NoWhereDashboardButton);
