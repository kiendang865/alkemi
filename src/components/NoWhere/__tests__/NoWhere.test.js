import React from "react";
import toJson from "enzyme-to-json";

import ReportIcon from "@material-ui/icons/Report";

import NoWhere from "../NoWhere";
import NoWhereDashboardButton from "containers/NoWhere/DashboardButton";
import createShallow from "utils/tests-utils/createShallow";

describe("<NoWhere />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });

    beforeEach(() => {
        wrapper = shallow(<NoWhere />);
    });

    it("renders 1 ReportIcon", () => {
        expect(wrapper.find(ReportIcon).length).toEqual(1);
    });

    it("renders 1 BackDashboardButton", () => {
        expect(wrapper.find(NoWhereDashboardButton).length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
