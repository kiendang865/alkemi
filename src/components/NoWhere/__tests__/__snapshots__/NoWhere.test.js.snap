// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoWhere /> matches snapshot 1`] = `
<div
  className="NoWhere-nowhere-1"
>
  <div
    className="NoWhere-nowhereHeader-2"
  >
    <ReportIcon
      className="NoWhere-nowhereIcon-4"
    />
    <h1
      className="NoWhere-nowhereTitle-3"
    >
      Oops... It seems that you are lost!
    </h1>
  </div>
  <Connect(WithStyles(NoWhereDashboardButton)) />
</div>
`;
